### ISSUE TITLE
Tested on Stretch and RasPi 3B+?

### ISSUE THREAD

**Issue by glynhudson** [sabhiram/raspberry-wifi-conf#32]
Has anyone been able to successfully run this on a RasPi 3B+ running Stretch?

**Comment by stzoran1** [comment]
It seems it is not working with Stretch and RasPi 3B... I will try to fix that on my fork...

**Comment by russdx** [comment]
Did any one ever get this working on a pi zero w 
I tried your version as well stzoran1 but still no luck

**Comment by stzoran1** [comment]
I didn’t test it on Zero W. I had some strange issues on some other projects including Zero W so I am trying to avoid it. If you want me to help you please provide more details about issues you have.

**Comment by russdx** [comment]
Hello, thanks for the quick reply.

I am using a raspberry pi zero with a old version of stretch (probably about 2 years old now?)

Linux 4.9.59+ #1047 Sun Oct 29 11:47:10 GMT 2017 armv6l GNU/Linux

PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"


I first installed the original version which all went ok but when running "npm run-script provision" it threw quite a few errors to do with the dhcpcd

I then installed your version and ran  "npm run-script provision" and it installed fine no errors. But when running the server it just crashed saying 8.8.8.8 in use (had not run previously)

**Comment by stzoran1** [comment]
Ok, I see...
First of all, I strongly recommend you to install fresh copy of Raspbian to be sure Linux on your SD card is not corrupted. If you don't want to do that at least upgrade your Raspbian using:

`
sudo apt-get update

sudo apt-get upgrade

sudo apt-get dist-upgrade

`

Then you can check which process is using port 8888 by executing next command:

`
sudo apt install lsof

lsof -i :8888

`

If you don't need this process (which is using port 8888) just uninstall it and try running again. If you need it open config.json and change server port to 8000 or 8001 or something like that.

**Comment by russdx** [comment]
Hello

Fresh install would be a bit of a nightmare as i have this install highly configured for what i use it for (dot matrix controller). I can try upgrading though.

I did change the port to 8.8.8.1 and yours started fine. But I never see the ap, iv read all the comments about restarting the server etc..  but i never see it appear. Does seam to be some sort of dhcpcd issue. I did a bit of googling and alot of others have had simular issues with turning raspberry pi zeros into access point and most of the fixes including re moving the 'driver=nl80211' in one of the files. I'm wondering if this could be a simular issue?

I can also upload my .img if this helps you debug? 

Thanks :)

**Comment by stzoran1** [comment]
As I previously said R Pi Zero is acting strange sometimes and there is few versions out here by my experience. In order to debug it I would need exactly the same board you have. I tested this version on multiple R Pi 3 B boards and everything worked fine. I shipped some solutions to the client based on this library and they tested on their hardware with positive feedback.

If you see message that server is started then you should be switched to AP mode.... Have you tried scanning WiFi network using mobile phone or laptop?

**Comment by russdx** [comment]
I get this

Wifi is not enabled, Enabling AP for self-configure

AP is not enabled yet... enabling...
... dhcpcd server restarted!
ifconfig wlan0 down successful...
ifconfig wlan0 up successful...
... dnsmasq server restarted!
... AP Enable Success!

HTTP server running...


But no ap is available (checked via 2 mobile and 1 laptop)
I am able to get a ap using this project https://github.com/jasbur/RaspiWiFi so the device can certainly do it.

Not quite sure what I am doing wrong as it all appears to say its working. Is there some error logs i can check maybe?

**Comment by stzoran1** [comment]
Yes, you can check dhcpcd log and dnsmasq logs for errors. Also, please send me output of next command to check:

`sudo iwconfig wlan0`

**Comment by russdx** [comment]
will do

wlan0     IEEE 802.11  ESSID:off/any
          Mode:Managed  Access Point: Not-Associated   Tx-Power=31 dBm
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:on

**Comment by stzoran1** [comment]
Ok, it seems it is in AP mode but SSID is not broadcasted. Please try next... Open config.json and set SSID to RPiHub instead of Iot-Hub-Config and try again.

**Comment by russdx** [comment]
still runs but still no ap available (that i can see)

**Comment by stzoran1** [comment]
I will try to examine what is going on but don’t know when I will have time to do that.

**Comment by russdx** [comment]
no worries, appreciate the help :)

**Comment by stzoran1** [comment]
Good luck and please let us know what was the problem if you succeed.

**Comment by sabhiram** [comment]
Dude @stzoran1 - you are the best. Thank you for running point on this issue, I have been very busy with my new endeavor that I barely have time to play gardener for these old babies of mine. I came across all this activity and your PR as a result of my once every few months scan of my old projects, my apologies for being a more silent party, and thank you for all the help and troubleshooting that you have done so far!

**Comment by theruin0000** [comment]
still not working on RasPi 3B+?
